Test_Automation/variable/IPSEC_VPN_DANOS_Variables.py

```diff
@@ -0,0 +1,85 @@
+'security vpn ipsec ike-group vm1-ike ike-version 2',
+'security vpn ipsec ike-group vm1-ike proposal 1 dh-group 19',
+'security vpn ipsec ike-group vm1-ike proposal 1 encryption aes128gcm128',
+'security vpn ipsec ike-group vm1-ike proposal 1 hash sha2_512',
+'security vpn ipsec site-to-site peer R1ipsecvpnpeerIP',
+'security vpn ipsec site-to-site peer R1ipsecvpnpeerIP authenticat mode pre-shared-secret',
+'security vpn ipsec site-to-site peer R1ipsecvpnpeerIP authenticat pre-shared-secret test123',
+'security vpn ipsec site-to-site peer R1ipsecvpnpeerIP default-esp-group vm1-esp',
+'security vpn ipsec site-to-site peer R1ipsecvpnpeerIP ike-group vm1-ike ',
+'security vpn ipsec site-to-site peer R1ipsecvpnpeerIP local-address R1ipsecvpnlocalIP',
+'security vpn ipsec site-to-site peer R1ipsecvpnpeerIP tunnel 0',
+'security vpn ipsec site-to-site peer R1ipsecvpnpeerIP tunnel 0 local prefix R1ipsecvpnPeerLocalPrefix/24',
+'security vpn ipsec site-to-site peer R1ipsecvpnpeerIP tunnel 0 remote prefix R1ipsecvpnPeerRemotePrefix/24',
+]
+
+R2_interface_config = [
+'interfaces dataplane R2R1interface address INTERFACE1IP/24',
+'interfaces dataplane R2R3interface address INTERFACE2IP/24',
+]
+
+R2_protocol_config = [
+'protocols ospf area 0 network NW/24',
+'protocols ospf area 0 network NW/24',
+]
+
+R3_interface_config = [
+'interfaces dataplane R3R2interface address INTERFACE1IP/24',
+'interfaces dataplane R3H2interface address INTERFACE2IP/24',
+]
+
+R3_interface_tunnel_config = [
+'interfaces tunnel tun0 local-ip R3tunnelinterfaceLocal',
+'interfaces tunnel tun0 remote-ip R3tunnelinterfaceRemote',
+'interfaces tunnel tun0 encapsulation gre',
+]
+
+R3_protocol_config = ['protocols ospf area 0 network NW/24',]
+
+R3_ipsec_vpn_config = [
+'security vpn ike make-before-break',
+'security vpn ipsec esp-group vm3-esp proposal 1 encryption aes128gcm128',
+'security vpn ipsec esp-group vm3-esp proposal 1 hash null',
+'security vpn ipsec ike-group vm2-ike ike-version 2',
+'security vpn ipsec ike-group vm2-ike proposal 1 dh-group 19',
+'security vpn ipsec ike-group vm2-ike proposal 1 encryption aes128gcm128',
+'security vpn ipsec ike-group vm2-ike proposal 1 hash sha2_512',
+'security vpn ipsec site-to-site peer R3ipsecvpnpeerIP',
+'security vpn ipsec site-to-site peer R3ipsecvpnpeerIP authenticat mode pre-shared-secret',
+'security vpn ipsec site-to-site peer R3ipsecvpnpeerIP authenticat pre-shared-secret test123',
+'security vpn ipsec site-to-site peer R3ipsecvpnpeerIP default-esp-group vm3-esp',
+'security vpn ipsec site-to-site peer R3ipsecvpnpeerIP ike-group vm2-ike ',
+'security vpn ipsec site-to-site peer R3ipsecvpnpeerIP local-address R3ipsecvpnlocalIP',
+'security vpn ipsec site-to-site peer R3ipsecvpnpeerIP tunnel 0',
+'security vpn ipsec site-to-site peer R3ipsecvpnpeerIP tunnel 0 remote prefix R3ipsecvpnPeerRemotePrefix/24',
+'security vpn ipsec site-to-site peer R3ipsecvpnpeerIP tunnel 0 local prefix R3ipsecvpnPeerLocalPrefix/24',
+]
```

Comment:
If I understand correctly your configuration method is a bit convoluted - it will work (I originally didnt understand what the replace strings were about).

Have you thought about something like,

R3_interface_config = Create List
...      interfaces dataplane R3R2interface address ${INTERFACE1IP}/24
...      interfaces dataplane R3H2interface address ${INTERFACE2IP}/24

This will do the substitution inline here and there is no need to 'replace string' else where

Reply:
Hi Trumpton,
Your comments have given me a good thought. I will look into this on how it can be built.
Initially, we had the variable file which can be updated by the user before execution. It was a bit tricky and the user should be very careful while updating the variable file. But, later realized that it will be good to have like data-driven approach so that, the test data can be received from a file so that the user can update the data in the test data file based on the topology created. Hence, created a test data file and taken care of replacing the string with the received input file.
I will keep this open. for the next script, I will try the approach you have provided and take it to other scripts.

Reply:
Hi Trumpton, This input helped and it is handled now.